subsys/bluetooth/mesh/brg_cfg.h

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_SUBSYS_BLUETOOTH_MESH_BRG_CFG_H_
+#define ZEPHYR_SUBSYS_BLUETOOTH_MESH_BRG_CFG_H_
+
+#include <stdint.h>
+#include <sys/types.h>
+#include <zephyr/kernel.h>
+
+/** These are internal APIs. They do not sanitize input params. */
+
+enum brg_tbl_direction {
+	BRG_PROHIBITED = 0,
+	BRG_ADDR1_TO_ADDR2 = 1,
+	BRG_ADDR1_TF_ADDR2 = 2,
+	BRG_RFU_MAX = 3,
+};
+
+#define BRG_NETIDX_NOMATCH 0xFFFF
+
+/* One row of the bridging table */
+struct brg_tbl_row {
+	/* Direction of the entry in the bridging table
+	 * 0 - no entry,
+	 * 1 - bridge messages with src as addr1 and dst as addr2
+	 * 2 - bridge messages with src as addr1 and dst as addr2 and vice-versa
+	 */
+	uint32_t direction:8;
```

Comment:
Are zephyr sources expected to be backwards compatible with switching between compilers? What is your expectation here?

Reply:
If I compile My App v1.0 with gcc and run it, set up a table, then compile v2.0 with clang, then DFU, I expect my briding table to load correctly after FW swap. I guess this is what Alex is getting at.

Reply:
I'm asking if this is a design goal we have for rest of the stack? I don't see this in RPL.c for example. It might be easier if reviewer states what is needed (like use `__attribute__((packed))` for compatibility). 

So, my question is, should we start using packing attribute or do something else?

Reply:
I'm note sure if `packed` guarantees anything about bitfields? My expectation across zephyr at least: bit layout in flash should never be compiler-dependent. This can even legally change between versions of the same compiler as far as I know.

Reply:
So, what is needed in this case: Don't use C bitfields in structs that will be written to persistent storage. Either pack bits manually using shifts and &, or use full integer types (leading to some extra flash usage).

Reply:
Can you clarify why is this a problem only in this case when some other module do this exactly same way (rpl.c, model pub in access.c, etc.)? 

Should rather this - compiler independent backwards compatibility - exercise be done globally at later point for all of mesh code?

Reply:
It is indeed a problem in those other modules as well as far as I can tell. I'm not strongly for or against postponing this to a separate task, I'll let @alxelax chime in since this was his concern originally

Reply:
Yes, this is a general ble mesh implementation issue in Zephyr. For instance, Zephyr's documentation claims it supports Arm Compiler: https://docs.zephyrproject.org/latest/develop/toolchains/arm_compiler.html
Should we continue implementation without bothering cross tool chain compatibility? Or should we stop using bitfields as data structure for storing data in persistent memory?
@carlescufi, probably you have any ideas?